packages/react/src/provider/use-open-feature-client-status.ts

```diff
@@ -28,6 +32,12 @@ export function useOpenFeatureClientStatus(): ProviderStatus {
       client.removeHandler(ProviderEvents.Reconciling, updateStatus);
     };
   }, [client]);
+  // highest priority > evaluation hook options > provider options > default options > lowest priority
+  const defaultedOptions = { ...DEFAULT_OPTIONS, ...useProviderOptions(), ...normalizeOptions(options) };
+
+  if (defaultedOptions.suspendUntilReady && status === ProviderStatus.NOT_READY) {
+    suspendUntilReady(client);
```

Comment:
Let me know if I'm doing this right @beeme1mr, I assume whenever the provider needs to do some asynciness, the status is NOT_READY.

Reply:
One thing to consider; we might also want to support suspense for this hook when the provider status is `RECONCILING`, since it's use-case more broad than `useWhenProviderReady`... Which would mean checking for that state too, and adding the related option(s).

@beeme1mr @aepfli @lukas-reining Any thoughts on this?

Reply:
I'd rather keep this hook clean without any suspense support. 

If we want to have suspense support we might want to extend the useWhenProviderReady hook to also return the current state additionally to the boolean.

Reply:
Ya, @wichopy I think I am starting to agree with @guidobrei on this one. I think I was wrong to attempt to add suspense here - I don't think it makes as much sense in this use-case - I think we should just remove the suspense and export the hook basically as is (though I think you've have too absorb some changes from main)

Sorry about the back-and-forth.

Reply:
done, please PTAL when you get another chance